src/app/ReadClient.h

```diff
@@ -134,8 +134,9 @@ class ReadClient : public Messaging::ExchangeDelegate
          * receives an OnDone call to destroy the object.
          *
          * @param[in] aSubscriptionId The identifier of the subscription that was established.
+         * @param[in] aMaxInterval The reporting max interval of the subscription that was established
          */
-        virtual void OnSubscriptionEstablished(SubscriptionId aSubscriptionId) {}
+        virtual void OnSubscriptionEstablished(SubscriptionId aSubscriptionId, uint16_t aMaxInterval) {}
```

Comment:
If we do decide to change the API (and see note about it not being needed for purposes of the Python bits) It might be worth passing in the ReadClient* instead, and the callee can get the max-interval, and whatever else it wants, from the ReadClient.  Otherwise we have to keep changing this API every time we discover that something else is needed.